File: 0169 majority-element/solution.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut res=0;
        let mut count=0;
        for i in 0..nums.len() {
            if count==0 {
                res=nums[i];
            }
            if res==nums[i]{
                count+=1;
            } 
            else {
                count-=1;
            }
        }
        res
    }
}

File: 0169 majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int c = 0,cd = 0;
        for(int i:nums) {
            if(c==0) {
                cd = i;
                c++;
            } else if(cd!=i) {
                c--;
            } else {
                c++;
            }
        }
        return cd;
    }
};
